### DISCUSSION TITLE
Custom cname resolver replacing vercel's `cname.vercel-dns.com.`

### DISCUSSION THREAD

**Discussion by HMJosh** [vercel/community#3607] [Feature Question] [answered]
### Summary

Hello! Got a question about networking for Vercel, using my Cloudflare domain, hope I can find help here

I'm creating a project that will need its own domain name assigned to each site.
for example: 
I get a customer and their site will be assigned to - site-random-123.domain.com
Then if I get another customer, their site will be assigned to site-random-456.domain.com

^^ I can do this easily myself with CF's API

Now I leverage the use of Vercel's domains API that creates a page to manage domains in the settings tab on the customer's website. But I cannot figure out how I go about setting whatever I need up to change Vercel's `cname.vercel-dns.com` to my own custom `cname.domain.com`

So basically keeping the setup the same for setting up custom domains to make it easy for customers to add their own custom domain name pointing to my infrastructure. Looked at Cloudflare for saas but I don't think that is what I need. I could be wrong

I'd like to keep my DNS host on Cloudflare. Any help is appreciated 

### Steps to Reproduce

not a bug. General question

**Comment by amyegan** [comment]
Hey @HMJosh. If the domain is using Cloudflare nameservers, then that is where you'll need to manage CNAME records. Another option is  using redirects or rewrites in [Middleware](https://vercel.com/docs/concepts/functions/edge-middleware) or [vercel.json](https://vercel.com/docs/concepts/projects/project-configuration#project-configuration/rewrites).

**Reply by HMJosh** [reply]
Thank you for the reply @amyegan, 

Maybe I didn't explain that properly
![ayikwoJ](https://github.com/vercel/community/assets/60105369/93508672-912a-4495-9d10-7d31ab5209a2)

I want to correctly change the required CNAME record the customer will put in their domain with my own.
so instead of them using `cname.vercel-dns.com.` I want them to use `cname.mycompany.com`

So it would look like this on the customers settings page to manage domains for their site:
![xLP4TtG](https://github.com/vercel/community/assets/60105369/610bd7cc-caea-43c6-8cc0-3721b07a4e42)

Is what you suggest achieving this outcome?

**Reply by amyegan** [reply, marked as answer]
That is the only CNAME record that will work to connect a domain or subdomain to a Vercel project. It cannot be customized.
